protocol/protos/NetRemoteWifi.proto

```diff
@@ -61,3 +61,39 @@ message WifiAccessPointDisableResult
     string AccessPointId = 1;
     WifiAccessPointOperationStatus Status = 2;
 }
+
+message WifiAccessPointSetPhyTypeRequest
+{
+    string AccessPointId = 1;
+    Microsoft.Net.Wifi.Dot11PhyType PhyType = 2;
+}
+
+message WifiAccessPointSetPhyTypeResult
+{
+    string AccessPointId = 1;
+    WifiAccessPointOperationStatus Status = 2;
+}
+
+message WifiAccessPointSetAuthenticationMethodRequest
+{
+    string AccessPointId = 1;
+    Microsoft.Net.Wifi.Dot11AuthenticationAlgorithm AuthenticationAlgorithm = 2;
```

Comment:
I know this mirrors the value specified in `AccessPointConfiguration`, so you've definitely done the right thing here. 

Looking forward, I think we need to expand this since the authentication method will require some additional data to configure it to work. For example, anything with a PSK will require the PSK to be specified and all such required configuration should all be provided together so the auth method can be atomically set from the client's perspective.

I'm not exactly sure how we would encode this, but at the very least, we probably need to wrap that configuration in separate messages/structures so they can evolve if necessary without needing to work hard to avoid breaking changes. 

Something like this could be one possible approach (just spitballing here):

```proto
message WifiAuthenticationMethodConfigurationPskPersonal
{
    Microsoft.Net.Wifi.SharedKey Psk;
    // ... whatever else is needed
}
message WifiAuthenticationMethodConfigurationEnterprise
{
    string RadiusServerAddress;
    // ... whatever else is needed
}
message WifiAuthenticationMethodConfiguration
{
   Microsoft.Net.Wifi.Dot11AuthenticationAlgorithm AuthenticationAlgorithm;
   oneof Configuration
   {
        WifiAuthenticationMethodConfigurationPskPersonal ConfigurationPskPersonal;
        WifiAuthenticationMethodConfigurationEnterprise ConfigurationEnterprise;
        // other auth method configuration structures ..
   }
   // ... whatever else is needed; other things can be safely added later without breaking backward compatibility.
}
```

We need to look at hostapd and see what it needs for each auth method, which can help guide how to encode that configuration here. I say "guide" instead of "define" because we shouldn't design only with hostapd in mind since that is a specific interpretation of how an AP is controlled. Anyway, check out this [hostapd.conf example file](https://w1.fi/cgit/hostap/tree/hostapd/hostapd.conf?h=hostap_2_10) which describes what each of the settings means. You can start with looking at the config args in this order of importance: `hw_mode`, `wpa`, `wpa_key_mgmt`, `wpa_pairwise`, `rsn_pairwise`, `group_cipher`, `group_mgmt_cipher`, `wpa_psk`, `wpa_passphrase`, `sae_password`, `wpa_psk_radius`. We would figure out which of these we need to set, then set it with the value(s) specified through the API using `Hostapd::SetProperty('config-property-name', 'property-value')`.

Reply:
Sounds good. I admit I didn't put too much thought into what was needed for the auth, I simply included the `Dot11AuthenticationAlgorithm` field, but it makes sense that certain auth methods need more information. I will take a look at hostapd and expand the API accordingly (the approach you gave looks pretty solid at first glance).